tools/packaging/kata-deploy/binary/src/artifacts/install.rs

```diff
@@ -0,0 +1,1397 @@
+    info!("deleting kata artifacts");
+
+    if Path::new(&config.host_install_dir).exists() {
+        fs::remove_dir_all(&config.host_install_dir)?;
+    }
+
+    nfd::remove_nfd_rules(config).await?;
+
+    if config.create_runtimeclasses {
+        runtimeclasses::delete_runtimeclasses(config).await?;
+    }
+
+    Ok(())
+}
+
+/// Note: The src parameter is kept to allow for unit testing with temporary directories,
+/// even though in production it always uses /opt/kata-artifacts/opt/kata
+fn copy_artifacts(src: &str, dst: &str) -> Result<()> {
+    for entry in WalkDir::new(src) {
+        let entry = entry?;
+        let src_path = entry.path();
+        let relative_path = src_path.strip_prefix(src)?;
+        let dst_path = Path::new(dst).join(relative_path);
+
+        if entry.file_type().is_dir() {
+            fs::create_dir_all(&dst_path)?;
+        } else {
+            if let Some(parent) = dst_path.parent() {
+                fs::create_dir_all(parent)?;
+            }
+
+            // Remove destination file first (ignore if it doesn't exist)
+            // This is crucial for atomic updates:
+            // - If the file is in use (e.g., a running binary), the old inode stays alive
+            // - The new copy creates a new inode
+            // - Running processes keep using the old inode until they exit
+            // - New processes use the new file immediately
+            match fs::remove_file(&dst_path) {
+                Ok(()) => {}                                             // File removed successfully
+                Err(e) if e.kind() == std::io::ErrorKind::NotFound => {} // File didn't exist, that's fine
+                Err(e) => return Err(e.into()), // Other errors should be propagated
+            }
+
+            fs::copy(src_path, &dst_path)?;
+        }
+    }
+    Ok(())
+}
+
+fn set_executable_permissions(dir: &str) -> Result<()> {
+    let bin_dir = Path::new(dir).join("bin");
+    if bin_dir.exists() {
+        for entry in fs::read_dir(&bin_dir)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_file() {
+                let mut perms = fs::metadata(&path)?.permissions();
+                perms.set_mode(0o755);
+                fs::set_permissions(&path, perms)?;
+            }
+        }
+    }
+
+    let runtime_rs_bin_dir = Path::new(dir).join("runtime-rs/bin");
+    if runtime_rs_bin_dir.exists() {
+        for entry in fs::read_dir(&runtime_rs_bin_dir)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_file() {
+                let mut perms = fs::metadata(&path)?.permissions();
+                perms.set_mode(0o755);
+                fs::set_permissions(&path, perms)?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn configure_shim_config(config: &Config, shim: &str) -> Result<()> {
+    let config_path = format!(
+        "/host/{}",
```

Comment:
what should the config_path be like? I am not clear about the path `/host/xxx` is for ? and I am afraid that the path "/host" is always valid in system.